### DISCUSSION TITLE
Migrating Database to New Server

### DISCUSSION THREAD

**Discussion by i-h8-github** [jeremyevans/rodauth#446] [General]
Hello,

I need to migrate a Rodauth user database from one server to another. Having followed the setup guide, my app user does not have permission for a lot of the tables and schemae, so pg_dump is not allowing me to backup the database. What is the recommended approach? Am I supposed to dump and restore table by table and user by user?

**Comment by jeremyevans** [comment]
If you followed the setup guide, you should have two users. You should be able to use 2 pg_dump commands, one as each user, to dump all of the tables.  If possible, I would dump and restore as the superuser, so you only need to use a single command.

**Reply by i-h8-github** [reply]
Yes, this was the way to do it. That's about what I figured, but I am skittish about anything regarding databases and authentication. Thank you for your encouraging response.
